connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java

```diff
@@ -203,7 +201,7 @@ private void setLevel(org.apache.logging.log4j.Logger logger, Level level) {
         LoggerConfig loggerConfig = context.getConfiguration().getLoggerConfig(loggerName);
         Level currentLevel = loggerConfig.getLevel();
 
-        if (level.equals(currentLevel)) {
+        if (lastModifiedTimes.containsKey(loggerName) && level.equals(currentLevel)) {
```

Comment:
why do we need this check? `lastModifiedTimes.containsKey(loggerName)`

Reply:
If we only check whether the level has changed, we'll run into the Log4j2 behavior change we mentioned before: all loggers will inherit the root logger's level if they haven't been pre-configured with a level. This means they won't appear in the `lastModified` map, and if they were already there, the update time won't change.
I originally thought users would expect to see entries in `lastModified` whenever `setLevel` is called, but this seems unnecessary. I'll fix this soon.

Reply:
Hi @chia7712,
I have noticed that the javadoc of `setLevel` states that the method will only return the loggers affected by this method.
IMHO, maybe we should keep the key check to prevent the output from containing null values.
WDYT?

Reply:
As we discussed off-line, the check of existence `lastModifiedTimes` can be removed as we should revise the test behaviour to match log4j2